src/service/private.overseas.entity.details.ts

```diff
@@ -0,0 +1,32 @@
+import { Session } from "@companieshouse/node-session-handler";
+import { Request } from "express";
+import { makeApiCallWithRetry } from "./retry.handler.service";
+import { createAndLogErrorRequest, logger } from "../utils/logger";
+import { OverseasEntityExtraDetails } from "@companieshouse/api-sdk-node/dist/services/overseas-entities/types";
+
+export const getPrivateOeDetails = async (
+  req: Request,
+  oeNumber: string,
+): Promise<OverseasEntityExtraDetails | undefined> => {
+  const response = await makeApiCallWithRetry(
+    "overseasEntity",
+    "getOverseasEntityDetails",
+    req,
+    req.session as Session,
+    oeNumber,
+  );
+
+  if (response.httpStatusCode !== 200 && response.httpStatusCode !== 404) {
+    const errorMsg = `Something went wrong fetching company details = ${JSON.stringify(response)}`;
```

Comment:
Do not print out the whole response. Just the status code & any error message.

Reply:
Log error for failure before throw.

Reply:
Why would we not want to log the `response` specifically in this instance?

And on line 21, the error is already being logged before it's thrown 👍

Reply:
error code would be enough here. Don't stringify JSON

Reply:
Going to leave it as is for consistency with the fetch in `company.profile.service.ts`.